holidays/calendars/chinese.py

```diff
@@ -1307,6 +1307,24 @@ class _ChineseLunisolar:
         2053: (FEB, 18),
     }
 
+    WINTER_SOLSTICE_THRESHOLDS: dict[str, dict[str, dict[int, int]]] = {
+        # UTC+7.
+        VIETNAMESE_CALENDAR: {
+            "dec23_thresholds": {3: 1943},
+            "dec21_thresholds": {0: 1980, 1: 2017, 2: 2050, 3: 2083},
+        },
+        # UTC+8.
+        CHINESE_CALENDAR: {
+            "dec23_thresholds": {3: 1947},
+            "dec21_thresholds": {0: 1988, 1: 2021, 2: 2058, 3: 2091},
+        },
+        # UTC+9.
+        KOREAN_CALENDAR: {
+            "dec23_thresholds": {3: 1955},
+            "dec21_thresholds": {0: 1992, 1: 2029, 2: 2062, 3: 2099},
```

Comment:
Any reason why dec23 goes before dec21?

Reply:
Only because it's used first in the code. :)